crates/compiler/can/src/scope.rs

```diff
@@ -429,7 +432,7 @@ impl Scope {
         self.aliases.contains_key(&name)
     }
 
-    pub fn inner_scope<F, T>(&mut self, f: F) -> T
+    pub fn inner_scope<F, T>(&mut self, entering_function: bool, f: F) -> T
```

Comment:
Consider splitting out into `enter_function`/`enter_binding` instead of a boolean here, since it might not obvious to a reader what true/false means at the call site.